src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -46,19 +61,23 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var err = obj as RazorError;
-            return (err != null) && Equals(err);
+            var error = obj as RazorError;
+            return Equals(error);
         }
 
         public override int GetHashCode()
         {
-            return base.GetHashCode();
+            return HashCodeCombiner.Start()
+                .Add(Message, StringComparer.Ordinal)
+                .Add(Location)
```

Comment:
separate issue if it is an issue at all.

BTW two errors with identical messages complaining about the same start point in the code probably _should_ be considered equal. so checking `Length` would just make `Equals()` less efficient.